### DISCUSSION TITLE
Current languages and translators

### DISCUSSION THREAD

**Discussion by m3nu** [borgbase/vorta#793] [Translations]
This thread is to coordinate the actual translation work.

Latest translation status: https://www.transifex.com/borgbase/vorta/

Current translators:

- Czech: Pavel Borecki
- Dutch: Heimen Stoffels
- French: @dbrassard and @lapineige
- Galician: Miguel Anxo Bouzada, [Transifex profile](https://www.transifex.com/user/profile/mbouzada/)
- German: @ThomasWaldmann, @Hofer-Julian, @XXXBold 
- Italian: @brokenpip3
- Polish: Tomasz Kontusz, @ktosiek
- Russian: Alexander Burchenko, [Transifex profile](https://www.transifex.com/user/profile/AABur/)
- Swedisch: Åke Engelbrektson, [Transifex profile](https://www.transifex.com/user/profile/eson/)

(let me know your Github and Transifex username to be added her)

Translations are organized on Transifex (TF) and then pulled into the source code, when 90+% complete. More details regarding translations are collected here: https://vorta.borgbase.com/contributing/translations/

Last, we should mention translators somewhere in the credits or the README. There will also be new strings added over time, as we're still adding features. So it's an ongoing commitment.

**Comment by dbrassard** [comment]
You can also use my GitHub handle

I am /u/dbrass from Reddit

**Comment by ThomasWaldmann** [comment]
There could be also a more general "CONTRIBUTORS" file, where everybody who contributed *something* (to code, to docs, to i18n, ...) can get added.

**Comment by ThomasWaldmann** [comment]
BTW, i noticed that transifex or the qt tools seem to just kill strings that do not completely match any more (or did I miss something?). So having the `*.ts` files in git is useful to c&p longer translated phrases and then just edit them as needed.

**Comment by m3nu** [comment]
Sometimes these services have a "translation memory". Even if the string was removed, it may suggest the removed translation for a similar string in the future. But yes, good to look it up on Github sometimes.

(Added GH handle for @dbrassard )

**Comment by m3nu** [comment]
> BTW, i noticed that transifex or the qt tools seem to just kill strings that do not completely match any more (or did I miss something?). So having the `*.ts` files in git is useful to c&p longer translated phrases and then just edit them as needed.

Those strings stay in "translation memory" and you can find them via Suggestions. I just played with replacing three dots with an ellipsis, but reverted back due to unicode-issues and your translations were still there.

<img width="676" alt="screen shot 2019-01-22 at 12 04 54" src="https://user-images.githubusercontent.com/3916435/51511903-2c353180-1e3e-11e9-90cf-82209ae5a287.png">

**Comment by ThomasWaldmann** [comment]
You had an exact match in that case.

What I meant are fuzzy matches - did these work for you also?

**Comment by m3nu** [comment]
Yes. Then it just gives you a e.g. 90% match and ranks it by match. So for each new string you get similar past matches.

I refactored many strings today and it was very helpful to keep the German translation updated.

**Comment by m3nu** [comment]
Added everyone who asked to join to the translation group. Short intro [here](https://github.com/borgbase/vorta/issues/159). But probably better to keep most discussion on Github.

Let me know how it goes, @dbrassard @brokenpip3. No tools are required to enter translations, but testing them needs Qt and the Transifex CLI, as described in [CONTRIBUTING.md > Translations](https://github.com/borgbase/vorta/blob/master/CONTRIBUTING.md#translations).

**Comment by dbrassard** [comment]
For strings with punctuation marks, should the translated text keep those? 
eg. "Repository Path:"

Also, the section "Data Flow to/from Transifex" is unclear to me. 

Finally, is there resources highlighting the proper workflow for Transifex? I can go on the website and translate strings but I need to test them in the final application. I installed the qt5 dev tools but I will still need some pointers (consider that I'm not a software developer).

**Comment by m3nu** [comment]
> For strings with punctuation marks, should the translated text keep those?
> eg. "Repository Path:"

Yes. Please keep all the punctuation marks including `...`, `:` and `.`

**Comment by m3nu** [comment]
> I need to test them in the final application. I installed the qt5 dev tools but I will still need some pointers (consider that I'm not a software developer).

There are a few ways to do this. Easiest is maybe:

1. download .ts file from Transifex
2. put it in `src/vorta/i18n/ts`
3. run `make translations-to-qm`
4. run `vorta` command

This assumes you already have Vorta installled as dev setup (editable installation from git). If not and you need further steps, let me know.

**Comment by brokenpip3** [comment]
Italian translation started.. give me a few days

**Comment by m3nu** [comment]
How are translations coming along? I see about half completed on Transifex. Good progress!

I was planning the next release when languages are done. Would be a few bug fixes and languages mostly.

**Comment by brokenpip3** [comment]
> How are translations coming along? I see about half completed on Transifex. Good progress!
> 

I'm near to finish (85%) it but i must review some choice.
I work on this a little time most days. I hope to finish the translations very soon.

**Comment by m3nu** [comment]
Just saw that Italian is now complete. Will we release French at the same time?

**Comment by dbrassard** [comment]
It is getting along. I'll have time to do some more translation today. Some strings just give me a bit of trouble : 
"nyx2.local-2018-11-16T09:49:58 from November 16, 2018". 

I will also need to test out the translation as I just used the Transifex interface until now. 

Do we need to validate/revise each translation in the Transifex interface?

**Comment by m3nu** [comment]
> "nyx2.local-2018-11-16T09:49:58 from November 16, 2018".

Looks like a placeholder. Either copy it or ignore and I'll remove it later.

The way you did it is fine. Just do "Save Changes". We don't use the review feature for now.

**Comment by dbrassard** [comment]
I finished the french translation today but I will need to test it in the application to spot mistakes. 

> This assumes you already have Vorta installled as dev setup (editable installation from git). If not and you need further steps, let me know.

I know how to pull a repository from git and push modifications but that's about it. What are the commands to build and install Vorta in dev mode? Do I need to rebuild the application completely when I switch the language file? It seems as I can modify the language of an already compiled program with the `\. make translations-to-qm`comand. 

P.S. I'm running Ubuntu 16.04

**Comment by ThomasWaldmann** [comment]
@dbrassard have a look in the vorta docs.

https://github.com/borgbase/vorta/blob/master/CONTRIBUTING.md

**Comment by m3nu** [comment]
This is amazing. I can't judge the actual translation, but it's fascinating that it works as expected. Your translations are already in the master-branch, which should make testing easier. I can also make some screenshots for you if that's quicker. In theory the steps are like this:

1. Checkout latest `master` branch or clone repo. `$ git clone https://github.com/borgbase/vorta`
2. Go to `vorta` root folder and run `$ make translations-to-qm` to compile translations.
3. Install editable version of Vorta: `$ pip install -e .`
4. Run Vorta with `$ vorta`. If you want to use a different language, use `$ LANG=it vorta`

**Comment by m3nu** [comment]
Some screenshots of the first tab and system tray menu:

<img width="1035" alt="screen shot 2019-02-13 at 15 50 34" src="https://user-images.githubusercontent.com/3916435/52695778-55159600-2fa8-11e9-92ca-4f8534567e7f.png">
<img width="1069" alt="screen shot 2019-02-13 at 15 50 04" src="https://user-images.githubusercontent.com/3916435/52695780-55159600-2fa8-11e9-82a2-5e67d3b498c4.png">

**Comment by brokenpip3** [comment]
Yesterday night i tried my translation, i found one error and 2-3 string to replace for better "standard word pick".
Today i replace them asap

**Comment by brokenpip3** [comment]
Ok fixed (i hope). (y)

**Comment by m3nu** [comment]
The new translations are now live as version 0.6.10. Please make sure it all shows correctly.

I'll pull new translations from Transifex before every release from now on, in case something needs changing.

**Comment by m3nu** [comment]
I'll soon add the latest translations for the next release. There are a few new strings that cover the new "Pass extra arguments to Borg" and "Improve compression options" functionality.

I already added translations for German. Maybe you guys can update FR and IT. @brokenpip3 @dbrassard 🙌🙌

Untranslated strings: https://www.transifex.com/borgbase/vorta/translate/#fr/vorta/165834924?q=translated%3Ano

**Comment by brokenpip3** [comment]
> I'll soon add the latest translations for the next release. There are a few new strings that cover the new "Pass extra arguments to Borg" and "Improve compression options" functionality.
> 
> I already added translations for German. Maybe you guys can update FR and IT. @brokenpip3 @dbrassard raised_handsraised_hands
> 
> Untranslated strings: https://www.transifex.com/borgbase/vorta/translate/#fr/vorta/165834924?q=translated%3Ano

I'm very sorry, i translated the new string the day after your post this message but forgot to reply here.

**Comment by m3nu** [comment]
No problem. I pulled all available translations for the release. So if they were on Transifex they should be included.

**Comment by m3nu** [comment]
Pinging @lapineige for French translation. Please join the Transifex group first and see here for other details: https://vorta.borgbase.com/contributing/translations/

**Comment by lapineige** [comment]
Request to join sent :)

**Comment by m3nu** [comment]
> Request to join sent :)

Already confirmed. 👍

**Comment by lapineige** [comment]
Already (almost) completely translated :)

**Comment by m3nu** [comment]
Added new Dutch translation and updated French translation.

**Comment by ktosiek** [comment]
Hi! I'd like to start a Polish translation. I am a native speaker. My Transifex login is tomasz.kontusz.

**Comment by ktosiek** [comment]
And a quick question to other translators: are there other translated borg frontends? Borg itself seems to not support translations at the moment, and BorgWeb only has a few words in French.

**Comment by m3nu** [comment]
Thanks for the offer, @ktosiek. I've added Polish on Transifex with you as first translator.

Let me know when done and we'll pull the files into Git, so Qt will load them.

**Comment by m3nu** [comment]
Added you for Finnish, @teemue. 👍

**Comment by m3nu** [comment]
Added Galician. Thanks to [Miguel Anxo Bouzada](https://www.transifex.com/user/profile/mbouzada/) 🙏

**Comment by m3nu** [comment]
As suggested in #774 by @Hofer-Julian, we will announce a "string freeze" here about 1-2 weeks before the next release. To give all translators enough time to add new strings on Transifex.

**Comment by m3nu** [comment]
Two new languages, Dutch and Galician fully translated by Heimen Stoffels and Miguel Anxo Bouzada respectively.

Will be in the next minor release.

**Comment by m3nu** [comment]
New version 0.7.3 is now out. It includes all new translations.

There will be a bugfix release soon after. So no problem to add new translations.

**Reply by lapineige** [reply]
Hello,

Regarding French translation, I translated most of the new (37/56) chains to translate, but I don't have time to finish it right now, so maybe it will need some love from someone else :)

ping @dbrassard maybe

**Comment by XXXBold** [comment]
I could do some translations into German (Native language) if needed, see some strings are still missing. Let me know if you're interested.

**Reply by m3nu** [reply]
Please go ahead. Translations sometimes change, so they are really ongoing. Just add yourself to Transifex. Latest strings will be there:

https://vorta.borgbase.com/contributing/translations/

**Comment by XXXBold** [comment]
I subscribed for transifex, but when I try to join the Team for translation, I couldn't select German from the list (didn't exist in control, but existed in the table below) I choosed Czech now because of that, hope this can be changed to German afterwards?

**Reply by m3nu** [reply]
Thanks for joining! I've added you to both languages. Hope it works now.

**Comment by XXXBold** [comment]
Yes it seems to work now. However, you should remove me from Czech, I don't speak it, just selected it so I could actually send a request on the site.

**Comment by adrianlopezgalera** [comment]
Hi. I want to collaborate to correct the Spanish translation in its entirety. I'm in Transifex with the alias @Julvenzor (https://app.transifex.com/user/profile/Julvenzor/), but I can't change anything because the chains appear, surprisingly, as "reviewed". I don't understand how someone can "reviewed" something that has some misspellings.

I am a writer in Spanish. I can understand that many people want to help translate and do not have the necessary language skills, but such individuals should check if what they write is correct.

In any case, I kindly request that I be granted the necessary permissions to correct the translation and add the missing strings.

Regards.

**Reply by real-yfprojects** [reply]
Is this resolved @Julvenzor?

**Reply by adrianlopezgalera** [reply]
No, it hasn't been resolved yet.

Thanks in advance.

**Reply by real-yfprojects** [reply]
I actually don't have the permissions to change your role. And reading [about roles](https://help.transifex.com/en/articles/6223416-understanding-user-roles) in the docs it seems like they are badly suited for a open source approach. Translators can only add translations to unreviewed strings while Reviewers can add translations and review translations. Though they can review their own translations. So while flawed, reviewer is the best role for our voluntary translators. 

@m3nu or @ThomasWaldmann  can one of you adjust @Julvenzors role?

For now I unreviewed those translations which you added an alternative suggestion to.

**Reply by adrianlopezgalera** [reply]
Thank you so much. As soon as I have some free time I will start reviewing the translation carefully while I have Vorta open to observe the context.

Regards.

**Reply by real-yfprojects** [reply]
> Vorta open to observe the context.

You might be able to download the translation files from transifex and open them with Qt Linguist which will show you some context information. However for some strings the context can only be found by looking into the source code.

**Reply by m3nu** [reply]
Sorry for the delay, guys. I've changed @real-yfprojects's role to admin and @Julvenzor to coordinator for Spanish. Let me know if everything works now.

**Reply by real-yfprojects** [reply]
Thanks @m3nu

**Reply by adrianlopezgalera** [reply]
Hello. Many thanks for everything. I feel very grateful for the trust placed in me. I'll check everything out in my free time.

Regards.

**Comment by adrianlopezgalera** [comment]
> > Vorta open to observe the context.
> 
> You might be able to download the translation files from transifex and open them with Qt Linguist which will show you some context information. However for some strings the context can only be found by looking into the source code.

Hello again, @real-yfprojects. I still don't have access to edit the translations. I haven't written to you during the week because I've been busy and I didn't want to bother you either. Sorry for the inconveniences.

Regards.

**Comment by jose1711** [comment]
After a very long time I finally upgraded Vorta to 0.9.1 only to find that Slovak is no more fully translated. I logged into Transifex only to find that it reports as 100% translated. The new strings are not there though so it seems that Transifex is not updating the source strings. Any idea? Has Transifex become abandoned? Thanks.

**Reply by m3nu** [reply]
No, Transifex is still used and I pull in translations before each release. Maybe the translation rate was below 85% back then?

**Reply by jose1711** [reply]
Thank you for your reply, here's a concrete example. In the current master (but also latest stable) there's a standalone "Recalculate" string:
https://github.com/borgbase/vorta/blob/master/src/vorta/assets/UI/archivetab.ui

The same string however is missing from both github `.ts` file (https://github.com/borgbase/vorta/blob/master/src/vorta/i18n/ts/vorta.sk.ts) and Transifex.

**Reply by m3nu** [reply]
There were some major issues with updating translations after we upgraded to PyQt6:

- The tool to extract translations change a good bit. Can't be used for now, since we can't mark custom translation functions.
- I tried the one from PySide6, but it add some garbage strings and garbage plurals that break Transifex.
- I'll try with the old PyQt5 tool again.

**Reply by m3nu** [reply]
I've reverted to the older PyQt5 tool for now. I extracts the strings we need. Including the ones you found missing:

https://app.transifex.com/borgbase/vorta/viewstrings/#en_US/vorta/489757532?q=text%3ARecalculate

**Reply by jose1711** [reply]
Thank you, it looks great!
